### PR TITLE
ssc-setup fixes

### PR DESCRIPTION
**Pull request by jwerle**
This PR improves `ssc setup` in various ways including the output during the FTE and setup to give users more feedback about what is going on aligning the output style with the rest of build scripts.

### CODE REVIEW

File: test/scripts/test-android-emulator.sh [hunk 1]
```diff
@@ -2,8 +2,23 @@
...
+
+if [[ -n $1 ]]; then
+  host=$1
+else
+  host="$(uname -s)"
+fi
+
+if [[ "$host" = "Linux" ]]; then
+  if [ -n "$WSL_DISTRO_NAME" ] || uname -r | grep 'Microsoft'; then
+    echo >&2 "erroR: WSL is not supported."
```
Comment (chicoxyzzy) [inline test/scripts/test-android-emulator.sh:14]:
```suggestion
    echo >&2 "ERROR: WSL is not supported."
```

File: bin/functions.sh [hunk 2]
```diff
@@ -387,8 +387,8 @@ function lower() {
   echo "$1"|tr '[:upper:]' '[:lower:]'
 }
 
-function version { 
-  echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; 
+function version() {
+  echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }';
 }
```
Comment (chicoxyzzy) [inline bin/functions.sh:392]: just curious, version of what?

**Review by chicoxyzzy — APPROVED** [review]
We need to document all the flags like `--only-env`, etc. It's hard to keep all this stuff in a head

**Comment by jwerle** [comment]
> We need to document all the flags like `--only-env`, etc. It's hard to keep all this stuff in a head

we'll come back and do a complete pass on docs for the scripts 🙏🏼

File: bin/android-functions.sh [hunk 3]
```diff
@@ -97,7 +100,7 @@ function test_javac_version() {
   fi
 
   jc_v="${jc_v/javac\ /}"
-  write_log "v" "Comparing $javac version: "$(version "$jc_v") "$(version "$target_version")"
+  write_log "v" "# Comparing $javac version: "$(version "$jc_v") "$(version "$target_version")"
   if [ "$(version "$jc_v")" -lt "$(version "$target_version")" ]; then
```
Comment (jwerle) [inline bin/android-functions.sh:104]: @chicoxyzzy

File: bin/functions.sh [hunk 4]
```diff
@@ -271,19 +271,19 @@ function prompt_new_path() {
...
-      write_log "h" "\"$input\" already exists, please choose a new path."
+      write_log "h" "# \"$input\" already exists, please choose a new path."
       if [ -n "$exists_message" ]; then
-        write_log "h" "$exists_message"
+        write_log "h" "# $exists_message"
       fi
       input=""
     elif [ -z "$input" ]; then
-      if prompt_yn "Cancel entering new path?"; then
+      if prompt_yn "> Cancel entering new path?"; then
```
Comment (mribbons) [inline bin/functions.sh:280]: prompt_yn calls prompt which outputs >, so this will prompt with >>

File: bin/functions.sh [hunk 5]
```diff
@@ -251,14 +251,14 @@ function prompt_new_path() {
...
-    if prompt_yn "Use suggested default path: ""$2""?"; then
+    if prompt_yn "> Use suggested default path: ""$2""?"; then
       input="$2"
     fi
   fi
 
   while true; do
     if [ -z "$input" ]; then
-      prompt "$text (Press Enter to go back)" input
+      prompt "> $text (Press Enter to go back)" input
```
Comment (mribbons) [inline bin/functions.sh:261]: prompt outputs >, so this will prompt with >>

File: src/cli/cli.cc [hunk 6]
```diff
@@ -114,6 +114,10 @@ String getSocketHome (bool verbose) {
   static String LOCALAPPDATA = getEnv("LOCALAPPDATA");
   static String SOCKET_HOME = getEnv("SOCKET_HOME");
   static String HOME = getEnv("HOME");
+  static const bool SSC_DEBUG = (
+    getEnv("SSC_DEBUG").size() > 0 ||
```
Comment (mribbons) [inline src/cli/cli.cc:118]: Nice. I think we also need to look at differentiating between debug output and debug builds in `install.sh` and `install.ps1`.

File: src/cli/cli.cc [hunk 7]
```diff
@@ -150,7 +164,11 @@ String getSocketHome () {
 }
 
 String getAndroidHome() {
-  auto androidHome = getEnv("ANDROID_HOME");
+  static auto androidHome = getEnv("ANDROID_HOME");
```
Comment (mribbons) [inline src/cli/cli.cc:167]: Using `static` here will make unit testing / mocking harder in future.

**Review by mribbons — CHANGES_REQUESTED** [review]

File: src/cli/cli.cc (hunk 7 above)
Comment (jwerle) [reply src/cli/cli.cc:167]: yeah I know what you mean, but let's punt on that for now until we can rethink this file

**Review by mribbons — APPROVED** [review]

**Pull request merged**